pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -597,15 +614,22 @@ func (dm *DockerManager) runContainer(
 		cpuShares = milliCPUToShares(cpuRequest.MilliValue())
 	}
 	var devices []dockercontainer.DeviceMapping
+	var nvidiaGPUPaths []string
+	nvidiaGPUPlugin := dm.GetNvidiaGPUPlugin()
 	if nvidiaGPULimit.Value() != 0 {
```

Comment:
Does the scheduler take GPU request / limits into account? If so, the kubelet should check that there is a GPU available in [`canAdmitPod`](https://github.com/kubernetes/kubernetes/blob/d92f685f14be868a71f25141245e0624234aaa02/pkg/kubelet/kubelet.go#L2198).